### PR TITLE
Alkha/main lfsparser

### PR DESCRIPTION
**Pull request by alkha-msft**
This is the end-to-end workflow of a basic caching system.

### CODE REVIEW

File: src/CloudLfs.Core.IntegrationTests/AzureTransferOrchestrationServiceTests.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+namespace CloudLfs.Core.UnitTests
+{
+    [TestClass]
+    public class AzureTransferOrchestrationServiceTests : TransferOrchestrationServiceBaseTests
+    {
+        public override ITransferOrchestrationService CreateTransferOrchestrationService(IConsoleBroker consoleBroker)
+        {
+            var messageService = new GitLfsMessageService(consoleBroker);
+            var blobBroker = new AzureBlobBroker(new Uri("https://cloudlfswus2premiumint.blob.core.windows.net/"));
```
Comment (vvicho) [inline src/CloudLfs.Core.IntegrationTests/AzureTransferOrchestrationServiceTests.cs:14]: should we use the non-premium one instead?

File: src/CloudLfs.Core.IntegrationTests/GitLfsBrokerTests.cs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+namespace CloudLfs.Core.UnitTests
+{
+    [TestClass]
+    public class GitLfsBrokerTests
+    {
+        [TestMethod]
+        public async Task GitLfsBroker_CanGetLfsObject222()
+        {
+            // authenticate
+            var authBroker = new GitBroker("D:\\Repos\\mixedreality.platform.v3");
```
Comment (vvicho) [inline src/CloudLfs.Core.IntegrationTests/GitLfsBrokerTests.cs:18]: we will probably need to make this a config or a temp folder, and if it doesn't exist then create it

Comment (vvicho) [reply src/CloudLfs.Core.IntegrationTests/GitLfsBrokerTests.cs:18]: actually, should this be removed? or are we trying to decide whether we specify the folder or use CurrentDirectory?

File: src/CloudLfs.Core/Brokers/AzureBlobBroker.cs [hunk 3]
```diff
@@ -1,49 +1,55 @@
...
+        }
+
+        public async Task<Response> DownloadAsync(string blobName, IProgress<long> progress, Stream contentStream, CancellationToken cancellationToken)
+        {
+            try
+            {
+                var blobClient = _containerClient.GetBlockBlobClient(blobName);
+                return await blobClient.DownloadToAsync(contentStream, new BlobDownloadToOptions { ProgressHandler = progress }, cancellationToken);
+            }
+            catch (RequestFailedException rfex)
```
Comment (vvicho) [inline src/CloudLfs.Core/Brokers/AzureBlobBroker.cs:35]: I think we can catch any exceptions on the AzureBlobService (TBD)

File: src/CloudLfs.Core/Services/TransferOrchestrationService.cs [hunk 4]
```diff
@@ -7,55 +7,105 @@
...
                 if (message is TerminateTransferGitLfsMessage)
                 {
-                    break;
+                    return;
                 }
                 else if (message is InitializeTransferGitLfsMessage)
                 {
                     _messageService.WriteMessage(new AcknowledgeGitLfsMessage());
                 }
+                else if (message is UploadObjectLfsGitMessage uploadMessage)
```
Comment (vvicho) [inline src/CloudLfs.Core/Services/TransferOrchestrationService.cs:42]: we can extract these into their own functions

**Review by vvicho — APPROVED** [review]

File: src/CloudLfs.Core/Brokers/AzureBlobBroker.cs (hunk 3 above)
Comment (alkha-msft) [reply src/CloudLfs.Core/Brokers/AzureBlobBroker.cs:35]: Will address in next PR.

File: src/CloudLfs.Core/Services/TransferOrchestrationService.cs (hunk 4 above)
Comment (alkha-msft) [reply src/CloudLfs.Core/Services/TransferOrchestrationService.cs:42]: Will address in next PR.

File: src/CloudLfs.Core.IntegrationTests/GitLfsBrokerTests.cs (hunk 2 above)
Comment (alkha-msft) [reply src/CloudLfs.Core.IntegrationTests/GitLfsBrokerTests.cs:18]: Will address in next PR. This should be a relative path.

File: src/CloudLfs.Core.IntegrationTests/AzureTransferOrchestrationServiceTests.cs (hunk 1 above)
Comment (alkha-msft) [reply src/CloudLfs.Core.IntegrationTests/AzureTransferOrchestrationServiceTests.cs:14]: Will address in next PR.

**Pull request merged**